File: ruby/lib/basket_actions/display_basket_command.rb
module BasketActions

	class DisplayBasketCommand

		def initialize(basket)
			@basket = basket
		end

		def run(cmd)
			items = @basket.list
			if items.empty?
				puts "Your basket is empty"
				return
			end
			items.each do |item|
				leader = sprintf("£%8.02f %5dx     ", item.price / 100.0, item.count)
				poss_len = leader.length + item.title.length
				title = (poss_len > 60) ? item.title[0, 60-leader.length-3]+"..." : item.title
				puts leader + title
			end
			total = basket_total(items)
			if total > 2000
				discount = total/10
				printf("£%8.02f            10%% discount\n", discount / -100.0)
				total -= discount
			end
			puts "---------------"
			printf("£%8.02f total\n", total / 100.0)
		end

		private

		def basket_total(items)
			total = 0
			items.each do |item|																# TODO -- inject
				total += item.price * item.count
			end
			total
		end

	end
end

File: python/basket_actions/display_basket_command.py
class DisplayBasketCommand:

    def __init__(self, basket):
        self.basket = basket

    def run(self, cmd):
        items = self.basket.list()
        if not items:
            print "Your basket is empty"
            return
        for item in items:
            leader = "$%8.02f %5dx     " % (item.price / 100.0, item.count)
            poss_len = len(leader) + len(item.title)
            title = item.title[0:60-len(leader)-3]+"..." if poss_len > 60 else item.title
            print leader + title
        total = self._basketTotal(items)
        if total > 2000:
            discount = total/10
            print "$%8.02f            10%% discount" % (discount / -100.0)
            total -= discount
        print "---------------"
        print "$%8.02f total\n" % (total / 100.0)

    def _basketTotal(self, items):
        total = 0
        for item in items:
            total += item.price * item.count
        return total

